### ISSUE TITLE
Checked out filter field (new feature)

### ISSUE THREAD

**Issue by nvyush** [joomla/joomla-cms#16959]
### Steps to reproduce the issue
We have command buttons 'Check-in' for lists of items (articles, contacts, categories etc.) at the admin panel, but we cannot filter checked out items.

I suggest to add that filter field (see the attached file) and add the key ```JOPTION_SELECT_CHECKED_OUT="- Select Checked out -"``` to the ```administrator/language/en-GB/en-GB.ini``` file. Then we can add to```filter_(...).xml``` files:
```
<field
	name="checked_out"
	type="checkedout"
	dbtable="_TABLE_NAME_"
	onchange="this.form.submit();"
>
	<option value="">JOPTION_SELECT_CHECKED_OUT</option>
	<option value="-1">JALL</option>
</field>
```

and to models files:
```
// Filter by checked_out
$checkedOut = $this->getState('filter.checked_out');

if (is_numeric($checkedOut))
{
	$checkedOut = (int) $checkedOut;
	$query->where('a.checked_out ' . (($checkedOut > 0) ? ' = ' . $checkedOut : ' > 0'));
}
```

### Additional comments

[checkedout.php.txt](https://github.com/joomla/joomla-cms/files/1121237/checkedout.php.txt)

**Comment by Bakual** [comment]
Since you already propose some code, can you please create a Pull request?
See https://docs.joomla.org/Using_the_Github_UI_to_Make_Pull_Requests

**Comment by nvyush** [comment]
@Bakual, should I place the file ```checkedout.php``` to the folder ```/libraries/cms/form/field``` or to the ```/libraries/joomla/form/fields```?

**Comment by Bakual** [comment]
I'd say to libraries/cms/form/field since it is CMS specific.

**Comment by nvyush** [comment]
@Bakual, I can't create file in the folder ```/libraries/cms/form/field```, I don't have access. Could you help me?

[new checkedout.php](https://github.com/joomla/joomla-cms/files/1176670/checkedout.php.txt)

**Comment by nvyush** [comment]
@Bakual, I prepare some model files (.php and filter_...xml) of many components (content, contact etc.) for PR, but I can't do PR without creating ```checkedout.php``` file. What I must do?

**Comment by Bakual** [comment]
You should be able to create files after you forked the repo. Then you work in your own instance of this repo and you have full access to do anything. Best to create a new branch first and then start doing your changes in that branch.

**Comment by nvyush** [comment]
The trouble is that the folders ```/libraries/cms/form``` and ```/libraries/cms/form/field``` are missing in my fork. I suppose, that I don't have access to do PR in that folder.

**Comment by brianteeman** [comment]
That means your forked repo is not up-to-date and you need to revise it

**Comment by nvyush** [comment]
I see this folders at my local installed copy of Joomla 3.7.4, but I don't see they at fork of staging branch.

**Comment by brianteeman** [comment]
yes - many files have been moved for the next release of Joomla

**Comment by nvyush** [comment]
@brianteeman, well, I try next release.

Could you look at pr #17316? The trouble is that com_content and com_contact use the same layout for categories (```/layouts/joomla/content/category_default.php```), but .xml-files of com_contact don't contain appropriate option ```show_category_heading_title_text```.

**Comment by nvyush** [comment]
Closed as we have PR #17491
